Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM python:3.12.0-bookworm as base
+    WORKDIR /app/todo-app/
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    ENV PATH="/root/.local/bin/:/app/todo-app/.venv/bin:${PATH}"
+
+
+FROM base as development
+    COPY poetry* /app/todo-app/
+    COPY pyproject.toml /app/todo-app/
+    RUN poetry install
+    ENTRYPOINT ["poetry", "run", "flask", "run"]
+
+FROM development as debug
+    ENTRYPOINT [ "tail", "-f", "/dev/null" ]
+
+FROM development as test
+    RUN apt update -qqy && apt install -qqy wget gnupg unzip
+    RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \  
+    && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list \  
+    && apt update -qqy \  
+    && apt -qqy install google-chrome-stable \  
+    && rm /etc/apt/sources.list.d/google-chrome.list \  
+    && rm -rf /var/lib/apt/lists/* /var/cache/apt/*
+    ENTRYPOINT ["poetry", "run", "pytest-watch", "--poll"]
+
+FROM base as production
+    COPY / /app/todo-app/
```

Comment:
Interesting, I've always seen this as:
```Dockerfile
    COPY . /app/todo-app/
```

But it looks like those are the same judging by:
> The <src> path must be inside the context of the build; you cannot COPY ../something /something, because the first step of a docker build is to send the context directory (and subdirectories) to the docker daemon.

Since that implies the build context is taken as the root of the folder structure available to copy
(taken from [here](https://docs.docker.com/engine/reference/builder/#copy))

No suggestion here, just thought this was interesting!